praw/models/user.py

```diff
@@ -117,7 +118,10 @@ def me(
     ) -> Optional[Redditor]:  # pylint: disable=invalid-name
         """Return a :class:`.Redditor` instance for the authenticated user.
 
-        In :attr:`~praw.Reddit.read_only` mode, this method returns ``None``.
+        In read-only mode this method returns ``None``, although you should
+        use :attr:`~praw.Reddit.read_only` to check if the client has a user
+        context. This may change in future to raise an error instead of
+        returning ``None``.
```

Comment:
```suggestion
        context. This behavior is deprecated, and will raise an exception
        in PRAW 8.
```
Instead of being vague, state the deprecation and the removal date.